### ISSUE TITLE
[Intermittent] The app is exited after reentering in immersive mode

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#2675]
**[Affected Versions]:**
- Firefox Reality version 8 RC1 (7a444d0).

**[Affected Platforms]:**
- Oculus Go - Android 7.1.2
- Oculus Quest- Android 7.1.1

**[Prerequisites]:**
- The headset is turned on and the browser is opened.

**[Steps to reproduce]:**
1. Play a 360 video in immersive mode.
2. Exit from immersive mode.
3. Reenter in immersive mode and observe the behavior.

**[Expected results]:**
- The video is successfully replayed in immersive mode.

**[Actual results]:**
- The app is exited.

**[Notes]:**
- In one attempt the issue was reproducible when first playing an 360 video in immersive mode.
- The previously opened websites are not restored after opening the app.
- I haven't have the chance to test this issue on the HTC devices but I will follow up with a comment on Monday.
- Attached is a screen recording of the issue: [link](https://drive.google.com/open?id=1vF-MDLWhxIf3sQWwU1tGBIzOTvUaDGQp).
- Attached is the logcat of the issue: [link](https://drive.google.com/open?id=13H7_yxBf5smWnyNXivN8HcSBGHoCHOO4).

**Comment by bluemarvin** [comment]
I can only reproduce this issue by entering and exiting 360 videos a lot. Eventually FxR gets low mem killed. I tried monitoring the various FxR processes: Main, Content, and Media, and none seemed to be growing. If the values of the memory monitor are accurate some system component may be leaking and causing memory problems.

**Comment by bluemarvin** [comment]
I'm starting to suspect the Oculus runtime is leaking every time we destroy/recreate a swap chain.

**Comment by bluemarvin** [comment]
I was not able to reproduce this issue on the Focus Plus using the same STR I used on the Quest to reproduce the issue multiple times. This is looking more like an Oculus Runtime bug.

@MortimerGoro can you try and reproduce this issue on the Quest? My guess is if we stop destroying and recreating the swap chain this might mitigate the issue for Oculus devices. I did notice that Youtube seems to work much better on the Focus than it does on the Quest. I wonder if we are getting bit by the CPU throttler?

**Comment by MortimerGoro** [comment]
@bluemarvin I was able to reproduce the issue, but only about 20 minutes of entering/exiting.

It seems like a leak in the Oculus VR SwapChain. Anyway I'll make sure there aren't any leaks in the swapChain creation in our code. There isn't any resize API for swapChains so I'm thinking on creating a pool in order to reuse them temporarily to reduce the amounts of create/reduce

**Comment by MortimerGoro** [comment]
I created a testcase that reproduces the issue using their SDK samples and contacted Oculus about it. Let's see what they say.

**Comment by bluemarvin** [comment]
@MortimerGoro Any updates from Oculus on this issue?

**Comment by emstanley** [comment]
@MortimerGoro have you heard anything else from Oculus on this?

**Comment by MortimerGoro** [comment]
@bluemarvin @emstanley I have checked the crash testcase with the latest runtime and it's fixed. Closing this one.  

@Softvision-RemusDranca you can test it again to confirm it's fixed.

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and it's no longer reproducible with the latest Firefox Reality version 10-141111916 (d9a7eec). I was not able to reproduce this issue in multiple attempts on multiple videos and I think it's safe to close it as verified fixed an open a new one if encountered in the future since is intermittent.
Tested on HTC Vive Focus (Android 7.1.2), Oculus Quest (Android 7.1.1) and Pico Neo 2 (Android - 8.1.0) devices.
